src/Servers/IIS/AspNetCoreModuleV2/CommonLib/Environment.cpp

```diff
@@ -188,16 +188,17 @@ void Environment::CopyToDirectoryInner(const std::filesystem::path& source, cons
                 }
             }
 
+            copiedFileCount++;
             CopyFile(path.path().wstring().c_str(), destinationPath.wstring().c_str(), FALSE);
         }
         else if (path.is_directory())
         {
             auto sourceInnerDirectory = path.path();
 
             // Make sure we aren't navigating into shadow copy directory.
-            if (sourceInnerDirectory.wstring().rfind(directoryToIgnore, 0) != 0)
+            if (!std::filesystem::equivalent(sourceInnerDirectory, directoryToIgnore))
```

Comment:
Is there any way to check if we are nested in the directoryToIgnore as well? That may be a bit more protective.

Reply:
That would be better, doesn't look like there is any nice api for it :(

Reply:
I was hoping a "startsWith" check would be useful here.

Reply:
Might be able to use https://en.cppreference.com/w/cpp/filesystem/path/compare, I'll have to test it a little to make sure it handles relative paths etc.

Reply:
I'm going back to the original code here, I think it is better since we make sure the path is absolute relative to the app path.